The condition is no longer set here, so just remove it.

Author: Alexander Grund (TU Dresden)

diff --git a/third_party/xla/third_party/hwloc/BUILD.system b/third_party/xla/third_party/hwloc/BUILD.system
--- a/third_party/xla/third_party/hwloc/BUILD.system
+++ b/third_party/xla/third_party/hwloc/BUILD.system
@@ -2,11 +2,6 @@
 
 licenses(["notice"])
 
-config_setting(
-    name = "with_numa_support",
-    define_values = {"with_numa_support": "true"},
-)
-
 filegroup(
     name = "COPYING",
     visibility = ["//visibility:public"],
@@ -14,9 +9,6 @@ filegroup(
 
 cc_library(
     name = "hwloc",
-    linkopts = select({
-        ":with_numa_support": ["-lhwloc"],
-        "//conditions:default": [],
-    }),
+    linkopts = ["-lhwloc"],
     visibility = ["//visibility:public"],
 )
